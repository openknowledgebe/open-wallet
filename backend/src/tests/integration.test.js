const { createTestClient } = require('apollo-server-testing');
const { auth } = require('./mocks/index');
const { db, models } = require('../');
const { constructTestServer, populate } = require('./utils');

const { LOGIN_ME_IN, GET_ME, REGISTER, ALL_USERS } = require('./graphql/queryStrings');

const testUser = { user: { name: 'Test Test', email: 'test@email.com', password: 'testing0189' } };

describe('Authenticated user', () => {
  let loggedUser;
  beforeAll(async () => {
    await db.connect();
    await populate();
    loggedUser = await models.User.findByEmail('john.doe@john.com');
  });

  afterAll(async () => {
    await db.disconnect();
  });

  describe('requires being logout', () => {
    let server;
    beforeAll(() => {
      server = constructTestServer({
        context: () => {
          return { models, user: loggedUser, auth };
        }
      });
    });
    it('to login', async () => {
      const { mutate } = createTestClient(server);
      const res = await mutate({
        mutation: LOGIN_ME_IN,
        variables: { email: 'john.doe@john.com', password: "doesn't matter" }
      });
      expect(res).toMatchSnapshot();
    });
    it('to register', async () => {
      const { mutate } = createTestClient(server);
      const res = await mutate({
        mutation: REGISTER,
        variables: testUser
      });
      expect(res).toMatchSnapshot();
    });
  });

  describe('GET_ME', () => {
    let server;
    beforeAll(() => {
      server = constructTestServer({
        context: () => {
          return { models, user: loggedUser, auth };
        }
      });
    });

    it('matches logged user', async () => {
      const { query } = createTestClient(server);
      const res = await query({
        query: GET_ME
      });
      expect(res.data.me).toEqual({ name: loggedUser.name, id: loggedUser.id });
    });
  });

  it('gets all users', async () => {
    const server = constructTestServer({
      context: () => {
        return { models, user: loggedUser, auth };
      }
    });

    const { query } = createTestClient(server);
    const res = await query({
      query: ALL_USERS
    });

    expect(res.data.users.length).toBeGreaterThanOrEqual(3);
  });
});
