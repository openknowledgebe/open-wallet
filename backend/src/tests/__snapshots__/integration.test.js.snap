// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticated user claims expenses succeeds 1`] = `
Object {
  "data": Object {
    "expenseClaim": Object {
      "flow": "IN",
      "type": "EXPENSE",
      "user": Object {
        "email": "john.doe@john.com",
        "name": "John Doe",
      },
    },
  },
}
`;

exports[`Authenticated user requires being logout to login 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: You must be logged out.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Authenticated user requires being logout to register 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: You must be logged out.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
